run_tests.py

```diff
@@ -29,47 +29,73 @@ def _makeResult(self):
     load_env_file(os.path.abspath(os.getcwd()) + os.sep + "environment.env")
     original_cwd = os.path.abspath(os.getcwd())
     os.chdir('.%stests%s' % (os.sep, os.sep))
-    suite = unittest.defaultTestLoader.discover('.', pattern="*.py")
+
+    # Run tests for selected module
+    if selected_module in ['API_CLI', 'API_QUERY', 'API_TASKS', \
+        'COMMON_CONFIGURATION', 'COMMON_DATE_PARSER', 'SENDER_CLI', \
+        'SENDER_NUMBER_LOOKUP', 'SENDER_SEND_DATA', 'SENDER_SEND_LOOKUP']:
+        dir_paths = {
+            'API_CLI': ['api', 'cli.py'],
+            'API_QUERY': ['api', 'query.py'],
+            'API_TASKS': ['api', 'tasks.py'],
+            'COMMON_CONFIGURATION': ['common', 'configuration.py'],
+            'COMMON_DATE_PARSER': ['common', 'date_parser.py'],
+            'SENDER_CLI': ['sender', 'cli.py'],
+            'SENDER_NUMBER_LOOKUP': ['sender', 'number_lookup.py'],
+            'SENDER_SEND_DATA': ['sender', 'send_data.py'],
+            'SENDER_SEND_LOOKUP': ['sender', 'send_lookup.py']
+        }
+        configured_path = '.' + os.sep + dir_paths[selected_module][0]
+        configured_pattern = dir_paths[selected_module][1]
+    else:
+        configured_path = '.'
+        configured_pattern = '*.py'
```

Comment:
**issue(blocking):** if the user passes a made-up module that is not in this list, we are assuming that all the tests should be running. I believe that we should raise an error instead.

Reply:
**suggestion:** instead of repeating the keys in both the list and the dict, you could place the dict as a constant at the top of the file (makes more sense to me) and then just do something like: 

```python
    if selected_module in DIR_PATHS.keys():
        ...
```

Reply:
**thought:** even if I understand the reasoning for this functionality, I believe that this is going to get more and more complicated over time and become unmaintainable. I have created an internal ticket for considering moving the testing suite to `pytest` which, among other facilities, provides a handy CLI, from which we could emulate this same functionality. Maybe it is time to start prioritizing that task (along with the Travis to GH Actions migration).